packages/core/src/fhir-deduplication/__tests__/group-same-immunizations.test.ts

```diff
@@ -1,58 +1,106 @@
     immunization.occurrenceDateTime = dateTime.start;
     immunization2.occurrenceDateTime = dateTime.start;
     immunization.vaccineCode = cvxCodeFlu;
     immunization2.vaccineCode = cvxCodeFlu;
 
-    const { immunizationsCvxMap } = groupSameImmunizations([immunization, immunization2]);
-    expect(immunizationsCvxMap.size).toBe(1);
+    const { immunizationsMap } = groupSameImmunizations([immunization, immunization2]);
+    expect(immunizationsMap.size).toBe(1);
   });
 
-  it("does not group immunizations with different dates", () => {
+  it("correctly groups immunizations with the same code, where one is missing the date", () => {
+    const immunization = makeImmunization();
+    const immunization2 = makeImmunization();
+
     immunization.occurrenceDateTime = dateTime.start;
-    immunization2.occurrenceDateTime = dateTime2.start;
     immunization.vaccineCode = cvxCodeFlu;
     immunization2.vaccineCode = cvxCodeFlu;
 
-    const { immunizationsCvxMap } = groupSameImmunizations([immunization, immunization2]);
-    expect(immunizationsCvxMap.size).toBe(2);
+    const { immunizationsMap } = groupSameImmunizations([immunization, immunization2]);
+    expect(immunizationsMap.size).toBe(1);
   });
 
-  it("removes immunizations with undefined dates", () => {
-    immunization.vaccineCode = cvxCodeFlu;
-    immunization2.vaccineCode = cvxCodeFlu;
+  it("correctly groups immunizations based on ndc codes", () => {
+    const immunization = makeImmunization();
+    const immunization2 = makeImmunization();
+
+    immunization.vaccineCode = ndcCodeFlu;
+    immunization2.vaccineCode = ndcCodeFlu;
 
-    const { immunizationsCvxMap } = groupSameImmunizations([immunization, immunization2]);
-    expect(immunizationsCvxMap.size).toBe(0);
+    const { immunizationsMap } = groupSameImmunizations([immunization, immunization2]);
+    expect(immunizationsMap.size).toBe(1);
   });
 
-  it("removes immunizations with unknown date", () => {
+  it("correctly groups immunizations based on codes, despite the order", () => {
+    const immunization = makeImmunization();
+    const immunization2 = makeImmunization();
+
+    immunization.vaccineCode = {
+      ...cvxCodeFlu,
+      coding: [...ndcCodeFlu.coding, ...cvxCodeFlu.coding],
+    };
+    immunization2.vaccineCode = {
+      ...cvxCodeFlu,
+      coding: [...cvxCodeFlu.coding, ...ndcCodeFlu.coding],
+    };
+
+    const { immunizationsMap } = groupSameImmunizations([immunization, immunization2]);
+    expect(immunizationsMap.size).toBe(1);
+  });
+
+  it("correctly groups immunizations based on display", () => {
+    const immunization = makeImmunization();
+    const immunization2 = makeImmunization();
+
+    const codeWithOnlyDisplay = {
+      ...cvxCodeFlu,
+      coding: [{ display: "Influenza, split virus, quadrivalent, PF" }],
+    };
+
+    immunization.vaccineCode = codeWithOnlyDisplay;
+    immunization2.vaccineCode = codeWithOnlyDisplay;
```

Comment:
We can make this cleaner (more "immutable") by:

```ts
    const vaccineCode = {
      ...cvxCodeFlu,
      coding: [{ display: "Influenza, split virus, quadrivalent, PF" }],
    };
    const immunization = makeImmunization({ vaccineCode });
    const immunization2 = makeImmunization({ vaccineCode });
```

It's test code, so not a big deal to update the object's properties (sometimes we need to do worse things to make tests work! 😅), but it's a good exercise and opportunity to discuss these patterns, and makes the code easier to understand.